app/src/main/java/org/mozilla/fenix/utils/StartupTaskManager.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.utils
+
+import mozilla.components.support.base.log.Log
+
+typealias StartupTask = () -> Unit
+
+/**
+ * A queue of tasks that are performed at specific points during Fenix startup.
+ *
+ * This queue contains a list of startup tasks. Each task in the queue will be started once Fenix
+ * is visually complete.
+ */
+class StartupTaskManager {
+    private var tasks = mutableListOf<StartupTask>()
+    internal var hasExecuted = false
+    private val startupTaskManagerQueueLogTag = "StartupTaskManager"
+
+    /**
+     * Add a task to the queue.
+     * Each task will execute on the main thread.
+     *
+     * @param task: The task to add to the queue.
+     */
+    @Synchronized
+    fun add(task: StartupTask) {
+        if (hasExecuted) {
+            Log.log(
```

Comment:
nit: remove this: the exception is logged to logcat so we don't need to log in addition.